src/discussions/discussions-home/DiscussionSidebar.jsx

```diff
@@ -1,17 +1,27 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 
 import classNames from 'classnames';
+import { useSelector } from 'react-redux';
 import {
   Redirect, Route, Switch, useLocation,
 } from 'react-router';
 
-import { Routes } from '../../data/constants';
+import { AppContext } from '@edx/frontend-platform/react';
+
+import { RequestStatus, Routes } from '../../data/constants';
 import { PostsView } from '../posts';
+import {
+  selectAllThreads, threadsLoadingStatus,
+} from '../posts/data/selectors';
 import { TopicsView } from '../topics';
 
 export default function DiscussionSidebar({ displaySidebar }) {
   const location = useLocation();
+  const AllThreads = useSelector(selectAllThreads);
```

Comment:
Please follow the style guide! Regular variables should not start with a capital.